compiler.go

```diff
@@ -203,6 +205,11 @@ func (c *Compiler) getSpecialVars(t *ast.Task, call *Call) (map[string]string, e
 		"ROOT_DIR":         c.Dir,
 		"USER_WORKING_DIR": c.UserWorkingDir,
 		"TASK_VERSION":     version.GetVersion(),
+		"CLI_ARGS":         flags.CliArgs,
+		"CLI_FORCE":        strconv.FormatBool(flags.Force || flags.ForceAll),
+		"CLI_SILENT":       strconv.FormatBool(flags.Silent),
+		"CLI_VERBOSE":      strconv.FormatBool(flags.Verbose),
+		"CLI_OFFLINE":      strconv.FormatBool(flags.Offline),
```

Comment:
We should not be using the `flags` package directly in the main `task` package. The CLI must be decoupled from the `Executor` so that they can be used independently.

These variables should also not be converted to strings. They should stay as booleans or we will break people's Taskfiles.